lib/core/src/resource_administration.cpp

```diff
@@ -0,0 +1,266 @@
+#include "resource_administration.hpp"
+
+#undef rxGeneralAdmin
+
+#ifdef IRODS_RESOURCE_ADMINISTRATION_ENABLE_SERVER_SIDE_API
+    #include "rsGeneralAdmin.hpp"
+
+    #define RODS_SERVER
+    #include "query_builder.hpp"
+    #undef RODS_SERVER
+
+    #define rxGeneralAdmin  rsGeneralAdmin
+#else
+    #include "generalAdmin.h"
+    #include "query_builder.hpp"
+
+    #define rxGeneralAdmin  rcGeneralAdmin
+#endif // IRODS_RESOURCE_ADMINISTRATION_ENABLE_SERVER_SIDE_API
+
+#include "rodsErrorTable.h"
+
+#include "fmt/format.h"
+
+namespace irods::experimental::administration::NAMESPACE_IMPL
+{
+    auto add_resource(RxComm& _comm, const registration_info& _info) -> std::error_code
+    {
+        const auto vault_path = fmt::format("{}:{}", _info.host_name, _info.vault_path);
```

Comment:
might need a slightly different name... 'location' within this scope could be good enough

Reply:
done.